tests/test_public.py

```diff
@@ -64,3 +66,45 @@ def test_equal_keys_are_equal(self):
     def test_different_keys_are_not_equal(self, k2):
         k1 = PrivateKey(b"\x00" * crypto_box_SECRETKEYBYTES)
         assert_not_equal(k1, k2)
+
+    def test_shared_key_getter(self):
+        """
+        RFC 7748 "Elliptic Curves for Security" gives a set of test
+        parameters for the Diffie-Hellman key exchange on Curve25519:
+
+        6.1.  [Diffie-Hellman on] Curve25519
+            [ . . . ]
+        Alice's private key, a:
+          77076d0a7318a57d3c16c17251b26645df4c2f87ebc0992ab177fba51db92c2a
+        Alice's public key, X25519(a, 9):
+          8520f0098930a754748b7ddcb43ef75a0dbf3a0d26381af4eba4a98eaa9b4e6a
+        Bob's private key, b:
+          5dab087e624a8a4b79e17f8b83800ee66f3bb1292618b6fd1c2f8b27ff88e0eb
+        Bob's public key, X25519(b, 9):
+          de9edb7d7b7dc1b4d35b61c2ece435373f8343c85b78674dadfc7e146f882b4f
+
+        Since libNaCl/libsodium shared key generation adds an HSalsa20
+        key derivation pass on the raw shared Diffie-Hellman key, which
+        is not exposed by itself, we just check the shared key for equality.
```

Comment:
Testing for equality isn't as robust as one would hope. The test vector of X25519 shared key passed through HSalsa20 is `1b27556473e985d462cd51197a9a46c76009549eac6474f206c4ee0844f68389`. Maybe something like this could be added here:

    To verify the test vector from NaCl source code, run
        $ wget https://hyperelliptic.org/nacl/nacl-20110221.tar.bz2
        $ bunzip2 < nacl-20110221.tar.bz2 | tar -xf -

    To view X25519 shared key is used as input
        $ cat nacl-20110221/tests/core1.c
    or
        $ head -8 nacl-20110221/tests/core1.c | tail -5

        Expected output
             0x4a,0x5d,0x9d,0x5b,0xa4,0xce,0x2d,0xe1
            ,0x72,0x8e,0x3b,0xf4,0x80,0x35,0x0f,0x25
            ,0xe0,0x7e,0x21,0xc9,0x47,0xd1,0x9e,0x33
            ,0x76,0xf0,0x9b,0x3c,0x1e,0x16,0x17,0x42

    To view the X25519 shared key passed through HSalsa20
        $ cat nacl-20110221/tests/core1.out

        Expected output
             0x1b,0x27,0x55,0x64,0x73,0xe9,0x85,0xd4
            ,0x62,0xcd,0x51,0x19,0x7a,0x9a,0x46,0xc7
            ,0x60,0x09,0x54,0x9e,0xac,0x64,0x74,0xf2
            ,0x06,0xc4,0xee,0x08,0x44,0xf6,0x83,0x89